src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py

```diff
@@ -526,17 +526,19 @@ def process_vnet_create_namespace(namespace):
         raise ValueError('incorrect usage: --subnet-name NAME [--subnet-prefix PREFIX]')
 
     if namespace.subnet_name and not namespace.subnet_prefix:
-        prefix_components = namespace.vnet_prefixes.split('/', 1)
+        if isinstance(namespace.vnet_prefixes, str):
+            namespace.vnet_prefixes = [namespace.vnet_prefixes]
+        prefix_components = namespace.vnet_prefixes[0].split('/', 1)
         address = prefix_components[0]
         bit_mask = int(prefix_components[1])
         subnet_mask = 24 if bit_mask < 24 else bit_mask
         namespace.subnet_prefix = '{}/{}'.format(address, subnet_mask)
 
 def process_vnet_gateway_create_namespace(namespace):
     ns = namespace
-    ns.enable_bgp = any([ns.asn or ns.bgp_peering_address or ns.peer_weight])
-    ns.create_client_configuration = any(ns.address_prefixes or [])
-    if ns.enable_bgp and not ns.asn:
+    validate_location(ns)
+    enable_bgp = any([ns.asn or ns.bgp_peering_address or ns.peer_weight])
```

Comment:
i think the `any` is not being used, rather it should be `any([ns.asn, ns.bgp_peering_address, ns.peer_weight])`?